KEP-3063: dynamic resource allocation: update kubelet plugin RPC versions supported. (#4164)
kubernetes/enhancements: keps/sig-node/3063-dynamic-resource-allocation/README.md

[PluginInfo](https://pkg.go.dev/k8s.io/kubelet/pkg/apis/pluginregistration/v1#PluginInfo)
response to distinguish the plugin from device and CSI plugins.

Under the advertised Unix Domain socket the kubelet plugin provides one of the
following supported gRPC interface versions. It was inspired by
[CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md),
with “volume” replaced by “resource” and volume specific parts removed.

Key difference between interface versions:

- [v1alpha2](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha2/api.proto)
interface provides resource claim information to kubelet plugin on one at a

> time basis. **NB! v1alpha2 will be deprecared, switch to v1alpha3**

Reviewer: What is this `NB!`?  It just renders as the letters NB here:
https://github.com/kubernetes/enhancements/pull/4164/files#diff-8fa237d276346416c2aafa209f721e43c9762f59cabec234eafafe01694de3ee

Author: An acronym signifying the importans, similar to `NOTE!` or `MIND THAT...`
https://en.wikipedia.org/wiki/Nota_bene

Reviewer: Let's remove it. I cant be alone in never seeing this acronym before and it doesn't seem to render as anything special in the markdown.

Author: Done.

> - [v1alpha3](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha3/api.proto)
> interface provides information about all resource claims of the pod, that belong
> to particular driver, so the kubelet plugin of this driver can consider all
> resources that need to be prepared or unprepared for the pod in one call.

Reviewer: ```suggestion
- [v1alpha3](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/dra/v1alpha3/api.proto)
interface provides information about all resource claims of a pod that belong
to a particular driver in a single call. This way the kubelet plugin of this driver can consider all
resources that need to be prepared or unprepared for the pod simultaneously.
```

Plugin SHALL assume that this RPC will be executed on the node where
the resource is being used.

> This RPC is called by kubelet when the last Pod using the resource is being
> deleted or has come to a final state ("Phase" is "done").

Reviewer: ```suggestion
This RPC is called by the kubelet when the last Pod using the resource is being
deleted or has reached a final state ("Phase" is "done").
```
